cli/install/cni-template.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2017 CNI authors
+// Modifications copyright (c) Linkerd authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file was inspired by:
+// 1) https://github.com/istio/cni/blob/c63a509539b5ed165a6617548c31b686f13c2133/deployments/kubernetes/install/manifests/istio-cni.yaml
+
+package install
+
+// CNITemplate provides the base template for the `linkerd install-cni-plugin` command.
+const CNITemplate = `### Namespace ###
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: {{.Namespace}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: linkerd-cni
+  namespace: {{.Namespace}}
+---
+# Include a clusterrole for the linkerd CNI DaemonSet,
+# and bind it to the linkerd-cni serviceaccount.
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: linkerd-cni
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "nodes", "namespaces"]
+    verbs: ["list", "get", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: linkerd-cni
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: linkerd-cni
+subjects:
+- kind: ServiceAccount
+  name: linkerd-cni
+  namespace: {{.Namespace}}
+---
+# This ConfigMap is used to configure a self-hosted linkerd CNI installation.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: linkerd-cni-config
+  namespace: {{.Namespace}}
+data:
+  incoming_proxy_port: "{{.InboundPort}}"
+  outgoing_proxy_port: "{{.OutboundPort}}"
+  proxy_uid: "{{.ProxyUID}}"
+  inbound_ports_to_ignore: "{{.IgnoreInboundPorts}}"
+  outbound_ports_to_ignore: "{{.IgnoreOutboundPorts}}"
+  simulate: "false"
+  log_level: "debug"
```

Comment:
should this be `log_level: "{{.LogLevel}}"`?

Reply:
Yes